debugtools/DDR_VM/src/com/ibm/j9ddr/corereaders/macho/ThreadCommand.java

```diff
@@ -81,6 +81,7 @@
 		ByteOrder endianness;
 		Map<String, Number> registers;
 
+		//  thread state names and size taken from <mach/i386/_structs.h>
```

Comment:
This comment is redundant given the change to line 55, and the 'size' aspect is now irrelevant.

Reply:
I think it's worth keeping track of where we can find the structure info for thread states, especially with Mac OSX on ARM being a case we may soon have to deal with.

The comment could be changed to `thread state structure info taken from <mach/machine/_structs.h>` to refer to the general header file which points to the architecture specific ones anyways.

Reply:
According to [1] the names are defined in the header file mention in the comment on line 55, not in `mach/machine/_structs.h` as this comment would have us believe. Please remove this misleading comment.

[1] https://opensource.apple.com/source/xnu/xnu-1504.7.4/osfmk/mach/i386/thread_status.h

Reply:
The thread state flavor names and constant values are found in `thread_status.h`, however, the thread state structure for each flavor is found in [_structs.h](https://opensource.apple.com/source/xnu/xnu-1504.7.4/osfmk/mach/i386/_structs.h), which is where the names for all the registers I used in the core reader comes from.

Also, I'm not sure how recent that source is, I've been looking at the headers included with Xcode, ie `/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/usr/include/mach/i386/thread_status.h`, and `thread_status.h` has some differences compared to the online source such as no `struct x86_saved_state64`. The thread commands, thread state flavors, and thread state structures looks to be the same though.

Reply:
There are three kinds of comments (in decreasing order of value):
* accurate comments (+1)
* missing comments (0)
* misleading comments (-1)

This seems to fall in the third category. Please either correct the comment or remove it. I'm fine if you want to say the 'structures' are defined in `_structs.h` (assuming that's true) but the state 'names' are not (else the comment on line 55 is wrong).

Reply:
Changed the comment to refer to the structure contents, hopefully it should be more clear now.

Reply:
Thank you.